src/opera/parser/tosca/csar.py

```diff
@@ -1,19 +1,27 @@
+import shutil
+import yaml
+
 from pathlib import Path
 from tempfile import TemporaryDirectory
 from zipfile import ZipFile
 
-import yaml
 from opera.error import ParseError
 
 
 class CloudServiceArchive:
-    def __init__(self, csar_name, csar_folder_path):
+    def __init__(self, csar_name, csar_folder_path=None):
```

Comment:
Why is this changed to be optional? For what you're using it, you need to pass the path as this parameter and provide another name - probaby the name of the folder itself.

Reply:
This `csar_folder_path` is used just in `validate_csar` function to as a directory (usually that is ".opera/csars") where tempdir for validation process is created, so I marked it here as unnecessary. I think I had in mind using it for something else but since it's only here it can be removed and also this line `with TemporaryDirectory(dir=self._csar_folder_path) as tempdir:` can be then just changed to `with TemporaryDirectory() as tempdir:` and things will still work as now (only the validation will be done in the tempdir outside of .opera).

Reply:
I have removed that unneeded option in a new separate commit.